force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js

```diff
@@ -147,19 +151,93 @@ export default class ServiceDeliveryRow extends LightningElement {
             });
     }
 
+    handleGetServicesForProgramEngagement(programEngagementId) {
+        this.isError = false;
+
+        getServicesByProgramEngagementId({ programEngagementId: programEngagementId })
+            .then(result => {
+                if (result) {
+                    this._filteredServices = result;
+                } else {
+                    this.isError = true;
+                }
+
+                this.enableServiceInputWithOptions();
+            })
+            .catch(err => {
+                this.rowError = handleError(err, false);
+            });
+    }
+
     handleInputChange(event) {
-        if (event.target.fieldName === this.fields.contact.fieldApiName) {
+        this.handleContactInputChange(event);
+        this.handleProgramEngagementInputChange(event);
+        this.handleServiceInputChange(event);
+    }
+
+    handleContactInputChange(event) {
+        if (event.target.fieldName !== this.fields.contact.fieldApiName) {
+            return;
+        }
+
+        if (
+            event.target.fieldName === this.fields.contact.fieldApiName &&
```

Comment:
Because are returning early if it is not the contact field, we can remove this check.